projects/angular-components/src/lib/collapse.directive.ts

```diff
@@ -0,0 +1,77 @@
+import { Directive, HostListener, Input, booleanAttribute, Output, EventEmitter, AfterViewInit } from '@angular/core';
+
+@Directive({
+  selector: '[collapse]',
+  standalone: true,
+})
+export class CollapseDirective implements AfterViewInit {
+  @Input({ required: true }) collapse!: HTMLElement;
+  @Input({ transform: booleanAttribute }) collapsed = false;
+  @Input() animationSpeed = 300;
+  @Output() collapsedChange = new EventEmitter<boolean>();
+
+  private collapseAnimation: Animation | null = null;
+  private clientHeight = 0;
+
+  ngAfterViewInit(): void {
+    this.setInitialState();
+  }
+
+  @HostListener('click')
+  onClick(): void {
+    this.collapsed = !this.collapsed;
+    this.collapsedChange.emit(this.collapsed);
+    this.updateState();
+  }
+
+  updateState(): void {
+    if (this.collapseAnimation) {
+      this.collapseAnimation.cancel();
+      return;
+    }
+
+    if (this.collapsed) {
+      this.clientHeight = this.collapse.clientHeight;
+      this.collapseElement();
+    } else {
+      this.expandElement();
+    }
+  }
+
+  private setInitialState(): void {
+    if (this.collapsed) {
+      this.clientHeight = this.collapse.clientHeight;
+      this.collapse.classList.add('collapsed');
```

Comment:
Since this library may be used in larger or heavier applications. I think it would be best to do custom rendering via Renderer2.

Reply:
What is the advantage of using renderer2 in this case?

Reply:
Keep in mind the element is already in the dom and managed by ElementRef